From b82e5e2ae143a0ce48ec800fbc9dfcf0a7c95311 Mon Sep 17 00:00:00 2001
From: Linn Crosetto <linn@hpe.com>
Date: Thu, 10 Mar 2016 18:25:03 -0700
Subject: [PATCH] xen/acpi: Disable ACPI table override when UEFI Secure Boot is
 enabled

Patch-mainline: Never, SUSE-Xen specific
References: bsc#970604

From the kernel documentation (initrd_table_override.txt):

  If the ACPI_INITRD_TABLE_OVERRIDE compile option is true, it is possible
  to override nearly any ACPI table provided by the BIOS with an
  instrumented, modified one.

Do not allow ACPI tables to be overridden if UEFI Secure Boot is enabled.

Signed-off-by: Linn Crosetto <linn@hpe.com>
Acked-by: Lee, Chun-Yi <jlee@suse.com> 
Automatically created from "patches.suse/acpi-Disable_ACPI_table_override_when_UEFI_Secure_Boot_is.patch" by xen-port-patches.py

--- a/arch/x86/kernel/setup-xen.c
+++ b/arch/x86/kernel/setup-xen.c
@@ -1269,6 +1269,13 @@ void __init setup_arch(char **cmdline_p)
 	/* Allocate bigger log buffer */
 	setup_log_buf(1);
 
+#ifdef CONFIG_EFI_SECURE_BOOT_SIG_ENFORCE
+	if (efi_enabled(EFI_SECURE_BOOT)) {
+		enforce_signed_modules();
+		pr_info("Secure boot enabled\n");
+	}
+#endif
+
 	reserve_initrd();
 
 #if defined(CONFIG_ACPI) && defined(CONFIG_BLK_DEV_INITRD)
@@ -1279,13 +1286,6 @@ void __init setup_arch(char **cmdline_p)
 
 	io_delay_init();
 
-#ifdef CONFIG_EFI_SECURE_BOOT_SIG_ENFORCE
-	if (efi_enabled(EFI_SECURE_BOOT)) {
-		enforce_signed_modules();
-		pr_info("Secure boot enabled\n");
-	}
-#endif
-
 #ifdef CONFIG_ACPI
 	if (!is_initial_xendomain()) {
 		printk(KERN_INFO "ACPI in unprivileged domain disabled\n");
